### PR TITLE
[WIP] Manage payment errors in ui

### PR DESCRIPTION
**Pull request by fgiobergia**

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a Pivotal story at the beginning of the PR title (see below).
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Remember to fix the PR title by removing WIP wording when ready
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/screens/wallet/payment/ConfirmPaymentMethodScreen.tsx [hunk 1]
```diff
@@ -51,27 +55,33 @@ import {
...
-  | Readonly<{
-      valid: false;
-    }>
-  | Readonly<{
-      valid: true;
-      wallet: Wallet;
-      amount: AmountInEuroCents;
-      fee: AmountInEuroCents;
-    }>;
+type ReduxMappedStateProps = Readonly<{
```
Comment (cloudify) [inline ts/screens/wallet/payment/ConfirmPaymentMethodScreen.tsx]: here you may want to have only "possible" states, e.g.:

```ts
type ReduxMappedStateProps =
    // loading
    | Readonly<{
       isLoading: true;
       error: None<string>;
       valid: false;
    }>
    | Readonly<{
        // completed with error
        isLoading: false;
        error: Some<string>;
        valid: false;
      }>
    | Readonly<{
        // completed with success
        isLoading: false;
        error: None<string>;
        valid: true;
        wallet: Wallet;
        amount: AmountInEuroCents;
        fee: AmountInEuroCents;
      }>;
```

File: ts/screens/wallet/payment/PickPspScreen.tsx [hunk 2]
```diff
@@ -38,19 +42,25 @@ import {
...
 import { buildAmount, centsToAmount } from "../../../utils/stringBuilder";
 
-type ReduxMappedStateProps =
-  | Readonly<{
-      valid: true;
-      pspList: ReadonlyArray<Psp>;
-    }>
-  | Readonly<{ valid: false }>;
+type ReduxMappedStateProps = Readonly<{
+  isLoading: boolean;
```
Comment (cloudify) [inline ts/screens/wallet/payment/PickPspScreen.tsx]: same as above, you want to define only possible states here

File: ts/screens/wallet/payment/TransactionSummaryScreen.tsx [hunk 3]
```diff
@@ -39,26 +42,32 @@ import {
...
-      valid: false;
-    }>
-  | Readonly<{
-      valid: true;
-      paymentReason: string | undefined;
-      paymentRecipient: EnteBeneficiario | undefined;
-      rptId: RptId | undefined;
-    }>;
+type ReduxMappedStateProps = Readonly<{
+  error: Option<string>;
```
Comment (cloudify) [inline ts/screens/wallet/payment/TransactionSummaryScreen.tsx]: here too

File: ts/types/errors.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  );
+  return maybeDetail.getOrElse("GENERIC_ERROR");
+};
+
+export const extractPaymentManagerError = (
+  _: PaymentManagerErrorType | undefined
+): PaymentManagerErrors => "GENERIC_ERROR";
+
+export const mapErrorCodeToMessage = (error: string): string => {
+  // TODO: create adequate messages & translate them
```
Comment (cloudify) [inline ts/types/errors.ts]: this is still to be done, so this story is still WIP

File: ts/types/errors.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    r => (r.status === 500 ? r.value.detail : undefined)
+  );
+  return maybeDetail.getOrElse("GENERIC_ERROR");
+};
+
+export const extractPaymentManagerError = (
+  _: PaymentManagerErrorType | undefined
+): PaymentManagerErrors => "GENERIC_ERROR";
+
+export const mapErrorCodeToMessage = (error: string): string => {
```
Comment (cloudify) [inline ts/types/errors.ts]: `error: PagoPaErrors` ?

**Comment by cloudify** [comment]
Superseded by #592

**Pull request closed without merging**